packages/nimble-components/src/table/models/data-hierarchy-manager.ts

```diff
@@ -0,0 +1,92 @@
+import { arrayToTree } from '../../utilities/thirdparty/performant-array-to-tree/arrayToTree';
+import type { TableNode, TableRecord } from '../types';
+import type { TableValidator } from './table-validator';
+
+/**
+ * Manages data hierarchy within the table, including converting between a flat list of
+ * data and hierarchical data.
+ */
+export class DataHierarchyManager<TData extends TableRecord> {
+    private isDataFlat = false;
```

Comment:
All of the data returned from the `getTableNodes` function has the `indexField` added on to it, so I don't think there is a trivial means of looking at the data and "knowing" that is flat or not. 

I'm not sure I'm following the suggestion, as this state is only of interest to this particular manager class (used in the `getAllRecords` implementation). If we returned this state, then it would become private state held on to the table and then passed into the `getAllRecords` method, which seems unnecessarily roundabout.

Reply:
See [this comment thread](https://github.com/ni/nimble/pull/1663#discussion_r1456538472).